src/mame/video/rad_hsvpal.h

```diff
@@ -0,0 +1,67 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood
+
+// this is based on emupal.h, style etc. (including use of statics) copied where possible.  If considered generic enough could probably be merged?
+
+#ifndef MAME_VIDEO_RAD_HSVPAL_H
+#define MAME_VIDEO_RAD_HSVPAL_H
+
+class radica_hsvpal_device : public palette_device
```

Comment:
This should probably derive from `device_t` and `device_palette_interface` rather than `palette_device`

Reply:
well I'm using the other palette_device functions, and wanted, from a driver code perspective, for it to act just like any other palette device.

I'm not actually convinced that emupal.cpp / emupal.h is good code, but I wanted to maintain compatibility with it.